components/edit_post_modal/edit_post_modal.jsx

```diff
@@ -55,15 +55,18 @@ export default class EditPostModal extends React.PureComponent {
             postError: '',
             errorClass: null,
             showEmojiPicker: false,
+            prevShowState: props.editingPost.show,
         };
     }
 
-    UNSAFE_componentWillUpdate(nextProps) { // eslint-disable-line camelcase
-        if (!this.props.editingPost.show && nextProps.editingPost.show) {
-            this.setState({
-                editText: nextProps.editingPost.post.message_source || nextProps.editingPost.post.message,
-            });
+    static getDerivedStateFromProps(props, state) {
+        if (props.editingPost.show && !state.prevShowState) {
```

Comment:
`state.prevShowState` is never set to true with how this is written, so `this.state.editText` is always updated as long as this visible. Could you fix it so that `state.prevShowState` is updated correctly based on `this.props.editingPost.show`?